File: Library/DTOs/Order/OrderDetails.cs
using Library.DTOs.Book;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Library.DTOs.Order
{
    public class OrderDetails
    {
        public Guid OrderId { get; set; }
        public float Price { get; set; }
        public DateTime Created {  get; set; }
        public OrderStatus Status { get; set; }
        public List<BookOrderDetails> OrderBooks { get; set; }
    }

    public class BookOrderDetails
    {
        public Guid BookId { get; set; }
        public string Title { get; set; }
        public string AuthorName { get; set; }
        public string ImageUrl {  get; set; }
        public float Price { get; set; }
        public int Amount { get; set; }

        //public CoverType CoverType { get; set; }
    }
}

File: client/src/types/types/order/OrderDetails.ts
import { CoverType } from "@/types/subTypes/book/CoverType";
import { Status } from "@/types/subTypes/Order/Status";

export interface OrderDetails{
    orderId:string;
    price: number;
    created: Date;
    status: Status;
    orderBooks: BookOrderDetails[]
}

export interface BookOrderDetails{
    bookId: string;
    title: string;
    authorName: string;
    price: number;
    imageUrl: string;
    amount: number;

    // coverType: CoverType;
}